Finder.php

```diff
@@ -134,7 +134,19 @@ public function in($path)
         }
 
         foreach ($path as $p) {
-            $this->_paths[] = $p;
+            if (1 === preg_match('/[\*\?\[\]]/', $p)) {
```

Comment:
Hum you are describing the behaviour of `preg_match_all`, `preg_match` only return `0`, `false` or `1`.

Relying on the tests to ensure that the pattern execution never became `false` is not valid for me because the tests will only described the known error cases. Maybe on a windows system, with specific character in a path the PCRE can fail and the function return `false`. It's why I've choose to use the `1` in test. This is the only valid answer which can make the code go to a `CallbackFilter` creation.

Maybe I'm missing a point in the test... But for me this is `1` which is the better value here 😅...